src/ParameterDistributions.jl

```diff
@@ -669,17 +669,15 @@ end
 
 function logpdf(pd::ParameterDistribution, xarray::AbstractVector{FT}) where {FT <: Real}
     #first check we don't have sampled distribution
-    for d in pd.distribution
-        if typeof(d) <: Samples
-            throw(
-                ErrorException(
-                    "Cannot compute logpdf of Samples distributions. Consider using a Parameterized type for your prior.",
-                ),
-            )
-        end
+    if any(isa.(pd.distribution, Samples))
+        throw(
+            ErrorException(
+                "Cannot compute logpdf of Samples distributions. Consider using a Parameterized type for your prior.",
+            ),
+        )
     end
     #assert xarray correct dim/length
-    if size(xarray)[1] != ndims(pd)
+    if length(xarray) != ndims(pd)
```

Comment:
`xarray` is an `AbstractVector`, so `size(xarray)[1]` is equivalent to its length.